### PR TITLE
pkg/debuginfo: Set debuginfod user-agent

### PR DESCRIPTION
**Pull request by brancz**
Fixes https://github.com/parca-dev/parca/issues/2914

### CODE REVIEW

File: pkg/parca/parca.go [hunk 1]
```diff
@@ -341,7 +343,10 @@ func Run(ctx context.Context, logger log.Logger, reg *prometheus.Registry, flags
 
 	var debuginfodClient debuginfo.DebuginfodClient = debuginfo.NopDebuginfodClient{}
 	if len(flags.Debuginfod.UpstreamServers) > 0 {
-		httpDebugInfoClient, err := debuginfo.NewHTTPDebuginfodClient(logger, flags.Debuginfod.UpstreamServers, flags.Debuginfod.HTTPRequestTimeout)
+		httpDebugInfoClient, err := debuginfo.NewHTTPDebuginfodClient(logger, flags.Debuginfod.UpstreamServers, &http.Client{
+			Transport: promconfig.NewUserAgentRoundTripper(fmt.Sprintf("parca-debugindod-%s", version), http.DefaultTransport),
```
Comment (kakkoyun) [inline pkg/parca/parca.go:347]:
```suggestion
			Transport: promconfig.NewUserAgentRoundTripper(fmt.Sprintf("parca-debugindod/%s", version), http.DefaultTransport),
```

I belive using `/` as version separator is more common https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent

Comment (brancz) [reply pkg/parca/parca.go]: fixed

**Review by kakkoyun — APPROVED** [review]

File: pkg/parca/parca.go [hunk 2]
```diff
@@ -341,7 +343,10 @@ func Run(ctx context.Context, logger log.Logger, reg *prometheus.Registry, flags
 
 	var debuginfodClient debuginfo.DebuginfodClient = debuginfo.NopDebuginfodClient{}
 	if len(flags.Debuginfod.UpstreamServers) > 0 {
-		httpDebugInfoClient, err := debuginfo.NewHTTPDebuginfodClient(logger, flags.Debuginfod.UpstreamServers, flags.Debuginfod.HTTPRequestTimeout)
+		httpDebugInfoClient, err := debuginfo.NewHTTPDebuginfodClient(logger, flags.Debuginfod.UpstreamServers, &http.Client{
+			Transport: promconfig.NewUserAgentRoundTripper(fmt.Sprintf("parca-debugindod-client/%s", version), http.DefaultTransport),
```
Comment (javierhonduco) [inline pkg/parca/parca.go:347]:
```suggestion
			Transport: promconfig.NewUserAgentRoundTripper(fmt.Sprintf("parca-debuginfod-client/%s", version), http.DefaultTransport),
```

I would suggest we add the project's URL so it's easier to find

**Review by javierhonduco — APPROVED** [review]

File: pkg/parca/parca.go (hunk 2 above)
Comment (kakkoyun) [reply pkg/parca/parca.go:347]: How did I miss that typo? 😅 I have just commented on that line. I need glasses 👓

Comment (brancz) [reply pkg/parca/parca.go:347]: oh my I've been making that mistake so often lately ....

**Pull request merged**